### PR TITLE
fix edge case in slice when start > end

### LINKED ISSUE #914: Wrong shape for circular Conv
```
import torch
import torch_xla
import torch.nn as nn
import pdb

device = 'xla:0'
# device='cpu'
m_cpu = nn.Conv1d(3, 4, 2, 2, (1,), 1, 1, True, 'circular')
m = nn.Conv1d(3, 4, 2, 2, (1,), 1, 1, True, 'circular').to(device)

input_cpu = torch.rand(2, 3, 5, requires_grad=True)
input = input_cpu.to(device).detach()
input.requires_grad = True

output_cpu = m_cpu(input_cpu)
output = m(input)
pdb.set_trace()

print(output_cpu.shape)
print(output.shape)
```

### PR DESCRIPTION
**Pull request by ailzhang**
fix #914

### CODE REVIEW

File: test/test_operations.py [hunk 1]
```diff
@@ -674,6 +674,14 @@ def test_norm_p0(self):
     xla_norm = xla_a.norm(p = 0)
     self.assertEqual(norm, xla_norm)
 
+  def test_slice_start_end(self):
```
Comment (dlibenzi) [inline test/test_operations.py]: We should use `runAtenTest()` as it reduces the amount of code:

https://github.com/pytorch/xla/blob/5067d0e34c06cf8a437942686daa48b0be62c0f4/test/test_operations.py#L719

File: test/test_operations.py [hunk 2]
```diff
@@ -161,7 +161,8 @@ def assertEqualRel(self, out, expected, rel_err=1e-2, abs_err=1e-5):
                               torch.ones_like(out).float() * abs_err)
       super(XlaTestCase, self).assertEqual(diff_tensor.size(),
                                            max_abs_err.size())
-      if torch.le(diff_tensor, max_abs_err).min().item() == 0:
+      if diff_tensor.numel() > 0\
```
Comment (dlibenzi) [inline test/test_operations.py]: Please do not use `\` to break Python lines (use parenthesis instead).

File: test/repro.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import torch
```
Comment (dlibenzi) [inline test/repro.py]: What is this file? :)

Comment (ailzhang) [reply test/repro.py]: ah lol accidentally checked in my small test file :P sorry

**Pull request merged**